bebraven/platform
# Allow deleting even if Canvas assignment is gone

### Description

**Tasks**
- https://app.asana.com/0/1168520191527013/1198984932600565
- https://app.asana.com/0/1168520191527013/1199091543014383

This PR allows you to delete an assignment in our DB (e.g., `BaseCourseCustomContentVersion`) even if the assignment in its `canvas_assignment_id` doesn't exist in Canvas for the course.

It was also more straightforward to do this in a transaction, so I refactored the controller code into the model as part of implementing this functionality.

### Motivation and Context

We can't control the Canvas UI and it's possible to delete the assignment from their website instead of our Course Management page. If that happens, you should be able to remove that assignment from our DB without needing a developer to go through the Rails console or DB directly. 

### Test

- Added specs
- Add a Project through the Course Management UI (_NEW PROJECT_)
<img width="1792" alt="Screen Shot 2020-11-06 at 7 46 53 PM" src="https://user-images.githubusercontent.com/62911934/98431324-276c5c80-2069-11eb-8fa6-271c9fbe7fcb.png">

- Go to the Canvas course and delete the corresponding assignment
- Click on the link from the Course Management UI and verify that the assignment no longer exists in Canvas
<img width="1792" alt="Screen Shot 2020-11-06 at 7 46 57 PM" src="https://user-images.githubusercontent.com/62911934/98431328-3226f180-2069-11eb-9cda-9ff5d9166fdb.png">

- Click **Delete** and see that the success message and that the project no longer appears in the list
<img width="1792" alt="Screen Shot 2020-11-06 at 7 47 04 PM" src="https://user-images.githubusercontent.com/62911934/98431333-3f43e080-2069-11eb-8cd5-2443d415af54.png">

### Types of changes




### Checklist:
